plugins/gssapi.c

```diff
@@ -1114,59 +1114,80 @@ gssapi_server_mech_authneg(context_t *text,
     gss_OID mech_type;
     gss_channel_bindings_t bindings = GSS_C_NO_CHANNEL_BINDINGS;
     struct gss_channel_bindings_struct cb = {0};
+    unsigned int rl; /* Required storage by getopt() */
+    const char *service_principal_opt = NULL; /* The configured service principal name */
+    unsigned char accept_any = 0;
+    const sasl_utils_t *utils = text->utils;
+#ifdef GSS_USE_LOAD_FROM
+    char* keytab = NULL; /* The configured credentials cache file path */
+
+    utils->getopt(utils->getopt_context, "GSSAPI", "keytab", (const char**)&keytab, &rl);
+#endif
 
     input_token = &real_input_token;
     output_token = &real_output_token;
     output_token->value = NULL; output_token->length = 0;
     input_token->value = NULL; input_token->length = 0;
-    
+
     if (text->server_name == GSS_C_NO_NAME) { /* only once */
-	if (params->serverFQDN == NULL
-	    || strlen(params->serverFQDN) == 0) {
```

Comment:
I acknowledge that this code was there already but if you're going to be making fixes in here:

Calling strlen() multiple times on the same string is just gross. Would prefer if it was only called once, with the return value saved if it's going to be referenced more than once. Also, checking `strlen(foo) == 0` is a stupid waste of a function call, should just be checking `*foo == '\0'` or just `!*foo`.

Reply:
Note that cleanup of the sort I suggest would be better in a separate commit. The current commit can be left as-is.